File: src/p1203/Main.java
package p1203;

import java.util.Scanner;

/**
 * Copyright(C) 2015,Cao Qingqing , Wuhan University All rights reserved.
 * 
 * This class is submitted for solving the problem 1203 in WOJ
 * 
 * @author : Qingqing Cao, Computer School, Wuhan University
 * 
 */
public class Main {

	public static void main(String args[]) {
		Scanner input = new Scanner(System.in);

		int num = 0;
		int loop = 0;
		int tmp = 0;
		int n = input.nextInt();
		while (n-- > 0) {
			tmp = input.nextInt();
			if (loop == 0) {
				loop++;
				num = tmp;
			} else {
				if (tmp == num)
					loop++;
				else
					loop--;
			}

		}
		System.out.println(num);
		input.close();
	}
}

File: src/p1203/Main.c
#include<stdio.h>
#include<stdlib.h>

int main()
{
    int no;
    while(scanf("%d",&no)==1)
    {
        int num=0,loop=0,tmp;
        while(no-->0)
        {
            scanf("%d",&tmp);
            if(loop==0)
            {
                loop++;
                num=tmp;
            }
            else
            {
                if(tmp==num) loop++;
                else loop--;
            }
        }
        printf("%d\n",num);
    }
    return 0;
}